### PR TITLE
Add --cprofile option to main and

### PR DESCRIPTION
**Pull request by rocky**
small styling stuff of misc files looked a in looking for preformance bottlnecks

### CODE REVIEW

File: mathics/main.py [hunk 1]
```diff
@@ -1,8 +1,16 @@
...
+Simpler Command-line interface to Mathics3.
+
+See also mathicsscript for a more sophisticated, and full
+featured CLI, which uses in more add-on Python packages and modules
+to assist in command-line behavior.
+"""
 
 import argparse
 import atexit
+import cProfile
```
Comment (mmatera) [inline mathics/main.py:13]: I just wonder, is this package available by default?

Comment (rocky) [reply mathics/main.py:13]: It is part of the standard library.

**Comment by mmatera** [comment]
Just one small comment: this that was put inside the CLI, could be implemented as a symbol, in the same way that  `Timing` works. Then, it would be available for the different front-ends

**Comment by rocky** [comment]
> Just one small comment: this that was put inside the CLI, could be implemented as a symbol, in the same way that `Timing` works. Then, it would be available for the different front-ends

Feel free to do or add on.

**Comment by mmatera** [comment]
> > Just one small comment: this that was put inside the CLI, could be implemented as a symbol, in the same way that `Timing` works. Then, it would be available for the different front-ends
> 
> Feel free to do or add on.

I mean, some thing like that:
```
class PythonProfileEvaluation(Builtin):
    """
    <dl>
      <dt>'PythonProfileEvaluation[$expr$]'
      <dd>profile $expr$
    </dl>
    """
    summary_text = "profile the internal evaluation of an expression"
    
    def eval(self, expr:Expression, evaluation:Evaluation):
        "PythonProfileEvaluation[expr_]"
        import cProfile
        profile_dict = SymbolNull
        if sys.version_info >= (3, 8):
            with cProfile.Profile() as pr:
                result = expr.evaluate(evaluation)
                # pr.create_stats()
                # profile_dict = from_python(pr.stats)
                # comment mm: It is a pitty that Profile.print_stats
                # does not have an interface to pass a stream, or
                # any other way to get a string instead of a direct
                # print to stdout...
                pr.print_stats()
        else:
            result = expr.evaluate(evaluation)
        return ListExpression(result, profile_dict)
       
```

**Comment by mmatera** [comment]
> > > Just one small comment: this that was put inside the CLI, could be implemented as a symbol, in the same way that `Timing` works. Then, it would be available for the different front-ends
> > 
> > 
> > Feel free to do or add on.
> 
> I mean, some thing like that:
> 
> ```
> class PythonProfileEvaluation(Builtin):
>     """
>     <dl>
>       <dt>'PythonProfileEvaluation[$expr$]'
>       <dd>profile $expr$
>     </dl>
>     """
>     summary_text = "profile the internal evaluation of an expression"
>     
>     def eval(self, expr:Expression, evaluation:Evaluation):
>         "PythonProfileEvaluation[expr_]"
>         import cProfile
>         profile_dict = SymbolNull
>         if sys.version_info >= (3, 8):
>             with cProfile.Profile() as pr:
>                 result = expr.evaluate(evaluation)
>                 # pr.create_stats()
>                 # profile_dict = from_python(pr.stats)
>                 # comment mm: It is a pitty that Profile.print_stats
>                 # does not have an interface to pass a stream, or
>                 # any other way to get a string instead of a direct
>                 # print to stdout...
>                 pr.print_stats()
>         else:
>             result = expr.evaluate(evaluation)
>         return ListExpression(result, profile_dict)
>        
> ```

The only problem with this is that it is not a direct way to get the text printed by `Profile.print_stats()` in the form of a `str`, in a way to be able to return it, or to show it as a message instead sending it to the stdout.

**Comment by rocky** [comment]
> The only problem with this is that it is not a direct way to get the text printed by `Profile.print_stats()` in the form of a `str`, in a way to be able to return it, or to show it as a message instead sending it to the stdout.

I am sure there is a way to dig into the API to do these kinds of things. But let's leave that for some other time and some other PR.

**Pull request merged**